iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -1181,6 +1180,35 @@ public void preDropPartitions(org.apache.hadoop.hive.metastore.api.Table hmsTabl
     context.putToProperties(HiveMetaStoreClient.SKIP_DROP_PARTITION, "true");
   }
 
+  private static void validatePartitionSpec(SearchArgument sarg, Table icebergTbl) {
+    for (PredicateLeaf leaf : sarg.getLeaves()) {
+      TransformSpec transformSpec = TransformSpec.fromStringWithColumnName(leaf.getColumnName());
+      Types.NestedField column = icebergTbl.schema().findField(transformSpec.getColumnName());
+      PartitionField partitionColumn =
+          icebergTbl.spec().fields().stream().filter(pf -> pf.sourceId() == column.fieldId()).findFirst().get();
+
+      if (!(partitionColumn.transform() == null && transformSpec.transformTypeString() == null) &&
+          !partitionColumn.transform().toString().equalsIgnoreCase(transformSpec.transformTypeString())) {
+        throw new UnsupportedOperationException(
+            "Invalid transform for column: " + transformSpec.getColumnName() +
+                " Expected: " + partitionColumn.transform() +
+                " Found: " + transformSpec.getTransformType());
+      }
+    }
+  }
+
+  private static UnboundPredicate<Object> getPartitionPredicate(Table icebergTbl, PartitionData partitionData,
```

Comment:
could we replace `icebergTbl` with `schema` <-` icebergTbl.schema()` and make it last arg?

Reply:
done